dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/internal/DLCTransactionProcessing.scala

```diff
@@ -87,17 +88,19 @@ private[bitcoins] trait DLCTransactionProcessing extends TransactionProcessing {
         val updatedOpt = Some(dlcDb.copy(state = DLCState.Claimed))
         Future.successful(updatedOpt)
       } else {
-        val withState = dlcDb.updateState(DLCState.RemoteClaimed)
         if (dlcDb.state != DLCState.RemoteClaimed) {
+          val withState = dlcDb.updateState(DLCState.RemoteClaimed)
           for {
-            // update so we can calculate correct DLCStatus
-            _ <- dlcDAO.update(withState)
```

Comment:
Don't do the database update here. This will make the database update for the state atomic with the rest of the writing to the database.